runtime/lua/vim/lsp.lua

```diff
@@ -1280,7 +1280,7 @@ function lsp.formatexpr(opts)
       local response =
         client.request_sync(ms.textDocument_rangeFormatting, params, timeout_ms, bufnr)
       if response and response.result then
-        lsp.util.apply_text_edits(response.result, 0, client.offset_encoding)
+        lsp.util.apply_text_edits(response.result, bufnr, client.offset_encoding)
```

Comment:
This is a bit of a redundant fix and is not strictly necessary since `apply_text_edits` will resolve a `0` buffer number, but since we already have the buffer number available anyway we might as well use it.

Reply:
+1 for the change. See my comment below I think this is actually the right fix